runtime/src/accounts_db.rs

```diff
@@ -4230,6 +4230,23 @@ impl AccountsDb {
         result
     }
 
+    fn sort_storages(storages: &[SnapshotStorage]) -> Vec<&Vec<Arc<AccountStorageEntry>>> {
+        let mut m1 = Measure::start("");
+        let mut result = storages
+            .iter()
+            .map(|item| {
+                (
+                    item.first().map(|item| item.slot()).unwrap_or_default(),
+                    item,
+                )
+            })
+            .collect::<Vec<_>>();
+        m1.stop();
+        error!("storage_sort_us: {}", m1.as_us());
+        result.par_sort_unstable_by(|a, b| a.0.cmp(&b.0));
+        result.into_iter().map(|(_, item)| item).collect()
```

Comment:
I do not know which is faster: this version here (A), or something like this (B):

```rust
let mut result = storages.clone();
result.par_sort_unstable_by(|a, b| a.first().unwrap_or_default().slot().cmp(&b.first().unwrap_or_default().slot()));
```

I could see sorting in (A) being helped by already having the slot pulled out front, which I think could remove an indirection in the `sort_by` in (B)? But maybe the prefetcher would be able to hide that anyway? I'm not sure without profiling/looking at the IR.

Is this something that even makes a difference in time? Is it something you already checked?

Reply:
I didn't check this specifically. I did try a few iterations of changes to see what is fastest. I learned refs were better as the payload than a cloned arc. fwiw, I just completely refactored all this and it is back in progress again because of new requirements - namely that I leave a space for slots with no stores, but that are in the write cache and could be ancestors or roots.